src/tools/wasm-fuzz-lattices.cpp

```diff
@@ -35,14 +42,421 @@ using namespace analysis;
 // Helps printing error messages.
 std::string LatticeComparisonNames[4] = {
   "No Relation", "Equal", "Less", "Greater"};
-std::string LatticeComparisonSymbols[4] = {"?", "=", "<", ">"};
 
 uint64_t getSeed() {
   // Return a (truly) random 64-bit value.
   std::random_device rand;
   return std::uniform_int_distribution<uint64_t>{}(rand);
 }
 
+// Actually a pointer to `L::ElementImpl`, but we erase the type to avoid
+// getting into a situation where `L` satisfying `Lattice` or `FullLattice`
+// circularly requires that `L` satisfies `Lattice` or `FullLattice`. C++ does
+// not allow concepts to depend on themselves. Also make the pointer copyable to
+// satisfy that constraint on lattice elements.
+template<typename L>
+struct RandomElement : std::unique_ptr<void, void (*)(void*)> {
```

Comment:
To be sure I follow, this is a unique pointer to a void (!), and also has a destructor of type `void *(void*)`?

Reply:
Correct!